python_dwd/file_path_handling/file_index_creation.py

```diff
@@ -0,0 +1,84 @@
+""" file index creation for available DWD station data """
+from pathlib import PurePosixPath
+import re
+from functools import lru_cache
+import ftplib
+import pandas as pd
+
+from python_dwd.constants.access_credentials import DWD_PATH, DWD_SERVER
+from python_dwd.constants.metadata import ARCHIVE_FORMAT, STATID_REGEX
+from python_dwd.download.ftp_handling import FTP
+from python_dwd.enumerations.column_names_enumeration import DWDMetaColumns
+from python_dwd.enumerations.parameter_enumeration import Parameter
+from python_dwd.enumerations.period_type_enumeration import PeriodType
+from python_dwd.enumerations.time_resolution_enumeration import TimeResolution
+
+
+@lru_cache(maxsize=None)
+def create_file_index_for_dwd_server(parameter: Parameter,
+                                     time_resolution: TimeResolution,
+                                     period_type: PeriodType) -> pd.DataFrame:
+    """
+    Function (cached) to create a file index of the DWD station data. The file index
+    is created for an individual set of parameters.
+    Args:
+        parameter: parameter of Parameter enumeration
+        time_resolution: time resolution of TimeResolution enumeration
+        period_type: period type of PeriodType enumeration
+    Returns:
+        file index in a pandas.DataFrame with sets of parameters and station id
+    """
+    server_path = PurePosixPath(DWD_PATH) / time_resolution.value / \
```

Comment:
Use PosixPath generation with all strings. afterwards you can use PosixPath in a formatted string e.g.

Reply:
What do you mean by that? **server_path** will be of type PosixPath.

Reply:
I mean sth like that `PorePosixPath([DWD_PATH, time_resolution.value,...])`